src/promptflow/promptflow/_utils/tool_utils.py

```diff
@@ -129,3 +130,30 @@ def get_inputs_for_prompt_template(template_str):
 def get_prompt_param_name_from_func(f):
     """Get the param name of prompt template on provider."""
     return next((k for k, annotation in f.__annotations__.items() if annotation == PromptTemplate), None)
+
+
+def get_input_type_mapping_for_prompt_template(template_str):
+    """Get all tagged input type mapping from a template string"""
```

Comment:
We may add some examples in the docstring to help understand the logic here?

Reply:
good point, will add.